src/store/middlewares/account.js

```diff
@@ -220,6 +221,14 @@ const accountMiddleware = store => next => (action) => {
     case actionTypes.accountLoggedOut:
       store.dispatch(emptyTransactionsData());
       break;
+    case actionTypes.settingsUpdated: {
+      const tokensList = action.data.token && action.data.token.list;
+      const token = tokensList && Object.keys(tokensList)[0];
+      if (tokensList[token]) {
```

Comment:
Now changing other settings (e.g. network switcher) causes an error in console because `tokensList` is not defined then. 
```suggestion
      if (tokensList && tokensList[token]) {
```